src/spatialdata/_core/operations/vectorize.py

```diff
@@ -146,11 +150,86 @@ def to_polygons(
     ----------
     data
         The SpatialElement representing the geometries to approximate as 2D polygons/multipolygons.
-    target_coordinate_system
-        The coordinate system to which the geometries to consider should be transformed.
 
     Returns
     -------
     The approximated 2D polygons/multipolygons in the specified coordinate system.
     """
-    raise RuntimeError("Unsupported type: {type(data)}")
+    raise RuntimeError(f"Unsupported type: {type(data)}")
+
+
+@to_polygons.register(SpatialImage)
+@to_polygons.register(MultiscaleSpatialImage)
+def _(
+    element: SpatialImage | MultiscaleSpatialImage,
+) -> GeoDataFrame:
+    model = get_model(element)
+    if model in (Image2DModel, Image3DModel):
+        raise RuntimeError("Cannot apply to_polygons() to images.")
+    if model == Labels3DModel:
+        raise RuntimeError("to_polygons() is not yet implemented for 3D labels.")
+
+    # reduce to the single scale case
+    if isinstance(element, MultiscaleSpatialImage):
+        element_single_scale = SpatialImage(element["scale0"].values().__iter__().__next__())
+    else:
+        element_single_scale = element
+
+    gdf_chunks = []
+
+    def _vectorize_chunk(
+        chunk: np.ndarray,  # type: ignore[type-arg]
+        block_info: dict[int | None, Any] | None = None,
+    ) -> da.Array:
+        if block_info is not None:
+            (yoff, _), (xoff, _) = block_info[0]["array-location"]
+            gdf = _vectorize_mask(chunk)
+            gdf["chunk-location"] = str(block_info[0]["chunk-location"])
+            gdf.geometry = gdf.translate(xoff, yoff)
+            gdf_chunks.append(gdf)
+        return da.zeros(chunk.shape)
+
+    element_single_scale.data.map_blocks(_vectorize_chunk).compute()
+
+    gdf = pd.concat(gdf_chunks)
+    gdf = GeoDataFrame([_dissolve_on_overlaps(*item) for item in gdf.groupby("label")], columns=["label", "geometry"])
+    gdf.index = gdf["label"]
+
+    transformations = get_transformation(element_single_scale, get_all=True)
+
+    assert isinstance(transformations, dict)
+
+    return ShapesModel.parse(gdf, transformations=transformations.copy())
+
+
+def _region_props_to_polygons(region_props: RegionProperties) -> list[Polygon]:
+    mask = np.pad(region_props.image, 1)
+    contours = skimage.measure.find_contours(mask, 0.5)
+
+    polygons = [Polygon(contour[:, [1, 0]]) for contour in contours if contour.shape[0] >= 4]
```

Comment:
I would add a comment explaining the choice of 4.

Reply:
I added a comment in the code.
When the number of vertices is 3, the shape is basically a line (2 points + repetition of the first point), which would raise a Shapely error. This is why I added this filtering.
Though, we should never have a shape with less than three coordinates: even a single pixel would be converted into a square. I just added this filtering for safety, but I'm not sure if skimage can produce a shape with less than 3 coordinates.